design/cstor/v1/pool_expansion_by_expanding_disk.md

```diff
@@ -0,0 +1,145 @@
+---
+oep-number: 2
+title: Pool Expansion When Underlying Disk Expanded
+authors:
+  - "@mittachaitu"
+owners:
+  - "@vishnuitta"
+  - "@kmova"
+  - "@prateekpandey14"
+  - "@sonasingh46"
+editor: "@mittachaitu"
+creation-date: 2020-03-20
+last-updated: 2020-03-20
+status: Implementable
+---
+
+# Pool Expansion When Underlying Disk Expanded
+
+## Table of Contents
+
+- [Pool Expansion When Underlying Disk Expanded](#pool-expansion-when-underlying-disk-expanded)
+	- [Table of Contents](#table-of-contents)
+	- [Summary](#summary)
+	- [Motivation](#motivation)
+		- [Goals](#goals)
+		- [Non-Goals](#non-goals)
+	- [Proposal](#proposal)
+		- [User Stories](#user-stories)
+		- [Current Implementation](#current-implementation)
+		- [Proposed Implementations](#proposed-implementations)
+		- [Steps to perform user stories](#steps-to-perform-user-stories)
+			- [Pool Expansion Steps:](#pool-expansion-steps)
+		- [Low Level Design](#low-level-design)
+			- [Work Flow](#work-flow)
+	- [Drawbacks](#drawbacks)
+	- [Alternatives](#alternatives)
+	- [Infrastructure Needed](#infrastructure-needed)
+
+## Summary
+
+- This proposal includes design details of the pool expansion when underlying disks were expanded.
+
+## Motivation
+
+There were cases where OpenEBS users expand the pool by expanding underlying disks(which were used by pool) when space on the pool got exhausted.
+
+### Goals
+
+- Expand cStor pool when the underlying disk got expanded.
+
+### Non-Goals
+
+NA
+
+## Proposal
+
+### User Stories
+
+As a OpenEBS user I should be able to expand the pool when underlying disk got expanded.
+
+### Current Implementation
+
+When user creates the CSPC, CSPC controller which is watching for CSPC will get create request and creats cstorpoolinstances and cstorpooldeployments. CstorPoolInstance is responsible for holding the blockdevice information and other pool configurations while cStor pool deployment contains following containers cstor-pool-mgmt, cstor-pool container and m-exporter as name conveyes cstor-pool-mgmt is responsible for managing the pool. cstor-pool container creates pool(ZFS file system) on top of the blockdevices. If any changes need to make on pool configuration like adding blockdevices, adding raid groups and replacing blockdevice cstor-pool-mgmt container will look into CstorPoolInstance for pool information and verifies are there any pending actions to be performed then cstor-pool-mgmt will prforms corresponding operation on the pool. Below info is hight level info to know how the operation is detected
```

Comment:
creates CStorPoolInstances and CStorPool deployments.

Reply:
cStor pool deployment have 3 containers named as `cstor-pool-mgmt`, `cstor-pool` and `m-exporter`

Reply:
`cstor-pool` container creates pool(ZFS file system) on top of the blockdevices, where as `cstor-pool-mgmt` sidecar container is responsible for managing these pool.

Reply:
If any changes required as day 2 operations on pool configuration like adding `BlockDevices`, adding raid groups and to replace `BlockDevices` , cstor-pool manager container will first verfiies any pending pool operations for CstorPoolInstance then performs the corresponding operation on the pool. 

Below is high level info to explain the operation is detected